e2e/Microsoft.Azure.Devices.E2ETests.NetStandard/ProvisioningTests.cs

```diff
@@ -0,0 +1,329 @@
+        }
+
+        [TestMethod]
+        [DataRow(nameof(ProvisioningTransportHandlerHttp), nameof(SecurityClientTpm), null, null)]
+        [DataRow(nameof(ProvisioningTransportHandlerHttp), nameof(SecurityClientX509), X509EnrollmentType.Individual, null)]
+        [DataRow(nameof(ProvisioningTransportHandlerHttp), nameof(SecurityClientX509), X509EnrollmentType.Group, null)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientTpm), null, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientTpm), null, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerAmqp), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientTpm), null, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.TcpOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientTpm), null, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Individual, TransportFallbackType.WebSocketOnly)]
+        [DataRow(nameof(ProvisioningTransportHandlerMqtt), nameof(SecurityClientX509), X509EnrollmentType.Group, TransportFallbackType.WebSocketOnly)]
+        public async Task ProvisioningDeviceClient_InvalidRegistrationId_Register_Fail(
+            string transportType,
+            string securityType,
+            X509EnrollmentType? x509EnrollmentType,
+            TransportFallbackType? transportFallback)
+        {
+            using (ProvisioningTransportHandler transport = CreateTransportHandlerFromName(transportType, transportFallback))
+            using (SecurityClient security = CreateSecurityClientFromName(securityType, x509EnrollmentType))
+            {
+                ProvisioningDeviceClient provClient = ProvisioningDeviceClient.Create(
+                    Configuration.Provisioning.IdScope,
+                    security,
+                    transport);
+
+                var cts = new CancellationTokenSource(PassingTimeoutMiliseconds);
+
+                //security.RegistrationID = "INVALID";
```

Comment:
>//security.RegistrationID = "INVALID"; [](start = 16, length = 38)

Looks like the test need to inject invalid RegistrationID (or bad cert or something) to cause the Registration failure.

Reply:
Fixed.